api/v1beta1/cinderapi_types.go

```diff
@@ -136,5 +144,8 @@ func init() {
 
 // IsReady - returns true if service is ready to serve requests
 func (instance CinderAPI) IsReady() bool {
-	return instance.Status.ReadyCount == *instance.Spec.Replicas
+	return instance.Generation == instance.Status.ObservedGeneration &&
+		instance.Status.ReadyCount == *instance.Spec.Replicas &&
+		(instance.Status.Conditions.IsTrue(condition.DeploymentReadyCondition) ||
```

Comment:
I can't tell whether L149 and L150 are truly necessary, or if they're already covered by L147 and L148. Are their situations where L147 and L148 are insufficient?

The same applies to the backup, sheduler and volume versions of the IsReady() function.

Reply:
There are cases were the generation and the ready count matches but we should not be ready.

For example if you just update the configuration of a service. In that case as soon as you enter the reconciliation loop we make the generation match, and because the replicas haven't changed this method would incorrectly return `true`.

I believe we need those 2 extra conditions to make sure we always return the right value.

Reply:
ack, that makes sense now.